conda_pypi/pre_command/extract_whl.py

```diff
@@ -0,0 +1,156 @@
+import codecs
+import json
+import os
+from os import PathLike
+
+from installer.records import RecordEntry, Hash
+from installer.sources import WheelFile
+from installer.destinations import WheelDestination
+from installer.utils import Scheme
+
+import installer.utils
+
+from typing import Literal, BinaryIO, Iterable, Tuple
+
+
+SUPPORTED_SCEMES: Tuple[Scheme] = ("platlib", "purelib")
+
+
+# inline version of
+# from conda.gateways.disk.create import write_as_json_to_file
+def write_as_json_to_file(file_path, obj):
+    with codecs.open(file_path, mode="wb", encoding="utf-8") as fo:
+        json_str = json.dumps(
+            obj,
+            indent=2,
+            sort_keys=True,
+            separators=(",", ": "),
+        )
+        fo.write(json_str)
+
+
+class MyWheelDestination(WheelDestination):
+    def __init__(self, target_full_path: str, source: WheelFile) -> None:
+        self.target_full_path = target_full_path
+        self.sp_dir = os.path.join(target_full_path, "site-packages")
+        self.entry_points = []
+        self.source = source
+
+    def write_script(
+        self, name: str, module: str, attr: str, section: Literal["console"] | Literal["gui"]
+    ) -> RecordEntry:
+        # TODO check if console/gui
+        entry_point = f"{name} = {module}:{attr}"
+        self.entry_points.append(entry_point)
+        return RecordEntry(
+            path="../../../bin/{name}",
```

Comment:
should this be an f-string?

```suggestion
            path=f"../../../bin/{name}",
```